sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java

```diff
@@ -159,13 +159,33 @@ void validate() {
         "Could not obtain Bigtable instance id");
   }
 
+  /**
+   * @param <T> parameter The runtime parameter.
+   * @return {@code null} if the input parameter is {@code null}, return the String value of runtime
+   *     parameter if the parameter is accessible, returns "Unavailable during pipeline
+   *     construction" if the runtime parameter is not accessible for debugging purpose.
+   */
+  @Nullable
+  static <T> String getDisplayDataForParameter(@Nullable ValueProvider<T> parameter) {
+    if (parameter == null) return null;
+
+    if (parameter.isAccessible()) {
+      return String.valueOf(parameter.get());
+    }
+    return "Unavailable during pipeline construction";
```

Comment:
This is error-prone. Please raise an error instead of returning a String here.

Reply:
This is expected as at pipeline construction time, runtime parameters don't have values yet and we don't want fail it.

I do think this is a bit misleading because parameter.isAccessible() could mean something else.  Any suggestion what we should do?

zhouyunqing@ can explain better.